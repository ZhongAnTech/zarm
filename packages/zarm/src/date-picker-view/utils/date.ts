export const getGregorianCalendar = (year, month, day, hour, minutes, seconds) => {
  return new Date(year, month, day, hour, minutes, seconds);
};

export const cloneDate = (date) => {
  return new Date(+date);
};

// 获取当月天数
export function getDaysInMonth(date) {
  return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
}

export const setMonth = (date, month) => {
  date.setDate(Math.min(date.getDate(), getDaysInMonth(new Date(date.getFullYear(), month))));
  date.setMonth(month);
};

// 补齐格式
export const pad = (n) => {
  return n < 10 ? `0${n}` : n;
};
