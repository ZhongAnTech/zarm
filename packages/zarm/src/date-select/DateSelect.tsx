import React, { PureComponent } from 'react';
import classnames from 'classnames';
import formatFn from '../date-picker-view/utils/format';
import DatePicker from '../date-picker';
import type { BaseDateSelectProps } from './PropsType';

export interface DateSelectProps extends BaseDateSelectProps {
  prefixCls?: string;
  className?: string;
}

export default class DateSelect extends PureComponent<DateSelectProps, any> {
  static defaultProps: DateSelectProps = {
    prefixCls: 'za-date-select',
    mode: 'date',
    disabled: false,
    minuteStep: 1,
    valueMember: 'value',
    hasArrow: true,
    onCancel: () => {},
  };

  static getDerivedStateFromProps(props) {
    return {
      selectValue: props.value,
    };
  }

  constructor(props) {
    super(props);
    this.state = {
      visible: false,
      selectValue: props.value,
    };
  }

  handleClick = () => {
    const { disabled } = this.props;
    if (disabled) {
      return false;
    }
    this.setState({
      visible: true,
    });
  };

  onChange = (selected) => {
    const { onChange } = this.props;
    if (typeof onChange === 'function') {
      onChange(selected);
    }
  };

  onOk = (selected) => {
    const { onOk } = this.props;
    this.setState({
      visible: false,
      selectValue: selected,
    });

    if (typeof onOk === 'function') {
      onOk(selected);
    }
  };

  onCancel = () => {
    const { onCancel } = this.props;
    this.setState({ visible: false });
    if (typeof onCancel === 'function') {
      onCancel();
    }
  };

  render() {
    const {
      prefixCls,
      className,
      placeholder,
      disabled,
      onChange,
      locale,
      value,
      ...others
    } = this.props;
    const { visible, selectValue } = this.state;

    const cls = classnames(prefixCls, {
      [`${prefixCls}--placeholder`]: !selectValue,
      [`${prefixCls}--disabled`]: disabled,
    });

    const arrowRender = <div className={`${prefixCls}__arrow`} />;

    return (
      <div className={cls} onClick={this.handleClick}>
        <input type="hidden" value={formatFn(this, selectValue)} />
        <div className={`${prefixCls}__input`}>
          <div className={`${prefixCls}__value`}>
            {formatFn(this, selectValue) || placeholder || locale!.placeholder}
          </div>
        </div>
        {arrowRender}
        <DatePicker
          {...others}
          className={className}
          visible={visible}
          value={selectValue}
          onOk={this.onOk}
          onCancel={this.onCancel}
        />
      </div>
    );
  }
}
