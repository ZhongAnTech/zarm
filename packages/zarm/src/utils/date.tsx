type DateType = 'y' | 'yyyy' | 'm' | 'mm' | 'd' | 'dd';
type DateOrString = Date | string;

export default {
  // 返回月份中的第一天是星期几
  firstDayOfMonth(date: DateOrString) {
    const d = date.constructor === Date ? date : this.cloneDate(date, 'dd', 1);
    return d.getDay();
  },
  // 获取当月天数
  getDaysInMonth(year: number, month: number) {
    return new Date(year, month, 0).getDate();
  },
  // 获取当月天数
  getDaysByDate(date: DateOrString) {
    const tmp = this.parseDate(date);
    return new Date(tmp.getFullYear(), tmp.getMonth() + 1, 0).getDate();
  },

  // 获取当月信息
  getCurrMonthInfo(year: number, month: number) {
    return {
      dayCount: new Date(year, month + 1, 0).getDate(), // 总天数
      firstDay: new Date(year, month, 1).getDay(), // 第一天是周几
    };
  },
  // 判断闰年还是平年
  isLeapYear(year: number | string) {
    if (year === '' || !Number.isSafeInteger(+year)) {
      throw new Error('年份格式不正确');
    }
    year = +year;
    if (year < 1790) {
      throw new Error('年份不能低于1790');
    }
    return (+year % 4 === 0 && +year % 100 !== 0) || +year % 400 === 0;
  },
  // 获取时间差的月份数
  getMonthCount(date1: DateOrString, date2: DateOrString) {
    const tmp1 = this.parseDate(date1);
    const tmp2 = this.parseDate(date2);
    const dur =
      (tmp2.getFullYear() - tmp1.getFullYear()) * 12 + (tmp2.getMonth() - tmp1.getMonth());
    return Math.abs(dur) + 1;
  },
  // 是否是今天(只判断年月日)
  isToday(date: DateOrString) {
    return this.isOneDay(date, new Date());
  },
  // 两个日期是否同一天
  isOneDay(date1: DateOrString, date2: DateOrString) {
    if (!date1 || !date2) {
      return false;
    }
    const tmp1 = this.parseDate(date1);
    const tmp2 = this.parseDate(date2);
    return tmp1.toDateString() === tmp2.toDateString();
  },
  // 两个日期是否同一个月
  isOneMonth(date1: DateOrString, date2: DateOrString) {
    if (!date1 || !date2) {
      return false;
    }
    const tmp1 = this.parseDate(date1);
    const tmp2 = this.parseDate(date2);
    return tmp1.getFullYear() === tmp2.getFullYear() && tmp1.getMonth() === tmp2.getMonth();
  },
  // 周几
  getDay(date: DateOrString, opt?: ReadonlyArray<string>) {
    const realDate = this.cloneDate(date);
    let array = ['周日', '周一', '周二', '周三', '周四', '周五', '周六'];
    if (opt && opt instanceof Array) {
      array = opt;
    }
    return array[realDate.getDay()];
  },
  // 标准化日期, 时间均为00:00:00
  parseDay(date: DateOrString) {
    const tmp = this.parseDate(date);
    return new Date(tmp.getFullYear(), tmp.getMonth(), tmp.getDate());
  },
  // 标准化时间
  parseDate(date: DateOrString | number): Date {
    if (date.constructor === Date) {
      return date;
    }
    if (date.constructor === String) {
      if (+date) {
        return new Date(+date);
      }
      date = date.replace(/-/gi, '/');
    }
    return new Date(date);
  },
  // 克隆日期
  cloneDate(date: DateOrString, type?: DateType, during?: number) {
    const tmp = new Date(this.parseDate(date));
    if (!type || !during) {
      return tmp;
    }
    switch (type) {
      case 'y':
        tmp.setFullYear(tmp.getFullYear() + during);
        break;
      case 'yyyy':
        tmp.setFullYear(during);
        break;
      case 'm':
        tmp.setMonth(tmp.getMonth() + during);
        break;
      case 'mm':
        tmp.setMonth(during);
        break;
      case 'd':
        tmp.setDate(tmp.getDate() + during);
        break;
      case 'dd':
        tmp.setDate(during);
        break;
      default:
        break;
    }
    return tmp;
  },
};
