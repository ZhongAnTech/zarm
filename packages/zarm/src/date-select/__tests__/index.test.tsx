import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DateSelect from '../index';
import enLocale from '../../date-picker-view/locale/en_US';
import { date1, date2, date3, date4, date5, date6 } from '../../../tests/testData/date';

describe('DateSelect', () => {
  it('DateSelect year', () => {
    const wrapper = mount(
      <DateSelect
        title="选择年份"
        placeholder="请选择年份"
        mode="year"
        value="2017"
        locale={enLocale}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect trigger visible', () => {
    const wrapper = mount(<DateSelect mode="date" defaultValue={date1} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect disabled', () => {
    const wrapper = mount(
      <DateSelect
        disabeld
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        value={date1}
      />,
    );

    wrapper.find('.za-date-select').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect date', () => {
    const wrapper = mount(
      <DateSelect title="选择日期" placeholder="请选择日期" mode="date" value={date1} />,
    );
    wrapper.setProps({ value: date1 });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect time', () => {
    const wrapper = mount(
      <DateSelect
        title="选择时间"
        placeholder="请选择时间"
        mode="time"
        defaultValue={date2}
        minuteStep={15}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect datetime', () => {
    const wrapper = mount(
      <DateSelect
        title="选择时间"
        placeholder="请选择时间"
        mode="datetime"
        min={date3}
        max={date4}
        defaultValue={date5}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ defaultValue: date6, value: date6 });
  });

  it('DateSelect wheelDefaultValue', () => {
    const wrapper = mount(
      <DateSelect
        title="选择日期"
        placeholder="请选择日期"
        mode="date"
        wheelDefaultValue={date5}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('should trigger onOk when press ok button', () => {
  //   const onOkFn = jest.fn();

  //   const wrapper = mount(
  //     <DateSelect
  //       mode="date"
  //       value="2009-3-4"
  //       onOk={onOkFn}
  //     />,
  //   );
  //   wrapper.find('.za-date-picker__submit').simulate('click');
  //   expect(onOkFn).toBeCalled();
  // });

  // it('should trigger onCancel when press cancel button', () => {
  //   const onCancelFn = jest.fn();

  //   const wrapper = mount(
  //     <DateSelect
  //       mode="date"
  //       value="2009-3-4"
  //       onCancel={onCancelFn}
  //     />,
  //   );

  //   wrapper.find('.za-date-picker__cancel').simulate('click');
  //   expect(onCancelFn).toBeCalled();
  // });

  // it('should trigger onMaskClick when click mask', () => {
  //   const onMaskClick = jest.fn();

  //   const wrapper = mount(
  //     <DateSelect
  //       mode="date"
  //       value="2009-3-4"
  //       visible
  //       onMaskClick={onMaskClick}
  //     />
  //   );

  //   wrapper.find('.za-mask').simulate('click');
  //   expect(onMaskClick).toBeCalled();
  // });
});
