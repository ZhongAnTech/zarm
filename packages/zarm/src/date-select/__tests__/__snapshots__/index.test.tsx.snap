// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelect DateSelect date 1`] = `
<ForwardRef
  disabled={false}
  hasArrow={true}
  minuteStep={1}
  mode="date"
  placeholder="请选择日期"
  title="选择日期"
  value={2019-04-23T00:00:00.000Z}
>
  <div
    className="za-date-select"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2019-04-23"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        2019-04-23
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    maskClosable={true}
    minuteStep={1}
    mode="date"
    onCancel={[Function]}
    onConfirm={[Function]}
    title="选择日期"
    value={2019-04-23T00:00:00.000Z}
    visible={false}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      title="选择日期"
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`DateSelect DateSelect datetime 1`] = `
<ForwardRef
  defaultValue={2017-11-03T10:00:00.000Z}
  disabled={false}
  hasArrow={true}
  max={2017-11-02T09:00:00.000Z}
  min={2017-11-02T03:00:00.000Z}
  minuteStep={1}
  mode="datetime"
  placeholder="请选择时间"
  title="选择时间"
>
  <div
    className="za-date-select za-date-select--placeholder"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="+0800-11-04 16:44"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        +0800-11-04 16:44
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    defaultValue={2017-11-03T10:00:00.000Z}
    maskClosable={true}
    max={2017-11-02T09:00:00.000Z}
    min={2017-11-02T03:00:00.000Z}
    minuteStep={1}
    mode="datetime"
    onCancel={[Function]}
    onConfirm={[Function]}
    title="选择时间"
    visible={false}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      title="选择时间"
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`DateSelect DateSelect disabled 1`] = `
<ForwardRef
  disabled={true}
  hasArrow={true}
  minuteStep={1}
  mode="date"
  value={2019-04-23T00:00:00.000Z}
>
  <div
    className="za-date-select za-date-select--disabled"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2019-04-23"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        2019-04-23
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    maskClosable={true}
    minuteStep={1}
    mode="date"
    onCancel={[Function]}
    onConfirm={[Function]}
    value={2019-04-23T00:00:00.000Z}
    visible={false}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`DateSelect DateSelect time 1`] = `
<ForwardRef
  defaultValue={2017-11-03T07:00:00.000Z}
  disabled={false}
  hasArrow={true}
  minuteStep={15}
  mode="time"
  placeholder="请选择时间"
  title="选择时间"
>
  <div
    className="za-date-select za-date-select--placeholder"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="16:44"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        16:44
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    defaultValue={2017-11-03T07:00:00.000Z}
    maskClosable={true}
    minuteStep={15}
    mode="time"
    onCancel={[Function]}
    onConfirm={[Function]}
    title="选择时间"
    visible={false}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      title="选择时间"
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`DateSelect DateSelect trigger visible 1`] = `
<ForwardRef
  defaultValue={2019-04-23T00:00:00.000Z}
  disabled={false}
  hasArrow={true}
  minuteStep={1}
  mode="date"
>
  <div
    className="za-date-select za-date-select--placeholder"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2022-11-04"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        2022-11-04
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    defaultValue={2019-04-23T00:00:00.000Z}
    maskClosable={true}
    minuteStep={1}
    mode="date"
    onCancel={[Function]}
    onConfirm={[Function]}
    visible={false}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`DateSelect DateSelect wheelDefaultValue 1`] = `
<ForwardRef
  disabled={false}
  hasArrow={true}
  minuteStep={1}
  mode="date"
  placeholder="请选择日期"
  title="选择日期"
  wheelDefaultValue={2017-11-03T10:00:00.000Z}
>
  <div
    className="za-date-select za-date-select--placeholder"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2022-11-04"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        2022-11-04
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    maskClosable={true}
    minuteStep={1}
    mode="date"
    onCancel={[Function]}
    onConfirm={[Function]}
    title="选择日期"
    visible={false}
    wheelDefaultValue={2017-11-03T10:00:00.000Z}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      title="选择日期"
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;

exports[`DateSelect DateSelect year 1`] = `
<ForwardRef
  disabled={false}
  hasArrow={true}
  minuteStep={1}
  mode="year"
  placeholder="请选择年份"
  title="选择年份"
  value="2017"
>
  <div
    className="za-date-select"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2017"
    />
    <div
      className="za-date-select__input"
    >
      <div
        className="za-date-select__value"
      >
        2017
      </div>
    </div>
    <div
      className="za-date-select__arrow"
    />
  </div>
  <ForwardRef
    maskClosable={true}
    minuteStep={1}
    mode="year"
    onCancel={[Function]}
    onConfirm={[Function]}
    title="选择年份"
    value="2017"
    visible={false}
  >
    <ForwardRef
      cancelText="取消"
      confirmText="确定"
      maskClosable={true}
      onCancel={[Function]}
      onClose={[Function]}
      onConfirm={[Function]}
      title="选择年份"
      visible={false}
    >
      <Popup
        destroy={true}
        direction="bottom"
        lockScroll={true}
        mask={true}
        onMaskClick={[Function]}
        visible={false}
      >
        <Trigger
          disabled={false}
          onClose={[Function]}
          visible={false}
        >
          <Mask
            color="black"
            destroy={true}
            mountContainer={[Function]}
            onClick={[Function]}
            opacity="normal"
            visible={false}
          >
            <Transition
              destroy={true}
              nodeRef={null}
              tranisitionName="za-fade"
              visible={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={true}
                nodeRef={
                  Object {
                    "current": undefined,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                unmountOnExit={true}
              />
            </Transition>
          </Mask>
          <Transition
            destroy={true}
            nodeRef={null}
            onEnter={[Function]}
            onLeaveEnd={[Function]}
            tranisitionName="za-move-up"
            visible={false}
          >
            <Transition
              addEndListener={[Function]}
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={true}
              nodeRef={
                Object {
                  "current": undefined,
                }
              }
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              unmountOnExit={true}
            />
          </Transition>
        </Trigger>
      </Popup>
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;
